dance/pipeline.py

```diff
@@ -523,11 +528,18 @@ def generate_config(
         See more detailed info from :meth:`generate`.
 
         """
+
         # Pre-checks
-        if pipeline is None and params is None:
-            raise ValueError("At least one of 'pipeline' or 'params' must be specified.")
-        elif pipeline is None and self.tune_mode == "pipeline":
-            raise ValueError("'pipeline' must be specified as tune mode is set to pipeline")
+        if pipeline is None and params is None and pipeline_params is None:
+            raise ValueError("At least one of 'pipeline' or 'params' or 'pipeline_params' must be specified.")
+        elif self.tune_mode == "pipeline":
+            if pipeline is None and pipeline_params is None:
```

Comment:
When tune_mode is set to pipeline, check whether pipeline exists. If not, check whether pipeline_params exists.